src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -96,9 +98,38 @@ protected string GetBaselineFileContents(string baselineFileName)
         string documentText, Range razorRange, bool isRazorFile, int hostDocumentSyncVersion = 0)
     {
         var codeDocument = CreateCodeDocument(documentText, isRazorFile, DefaultTagHelpers);
-        var csharpRange = GetMappedCSharpRange(codeDocument, razorRange);
         var csharpTokens = Array.Empty<int>();
 
+        if (UseRangesParams)
+        {
+            var csharpRanges = GetMappedCSharpRanges(codeDocument, razorRange);
```

Comment:
Again, seems like this code should be combined with the below, and just have that operate on a single element array.